### PR TITLE
PR: Fix regex quoting

### PR DESCRIPTION
**Pull request by juliangilbey**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

The regex that this PR fixes was not quoted correctly; this PR fixes the quoting.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @juliangilbey

### CODE REVIEW

**Review by ccordoba12 — COMMENTED** [review]
I don't understand this change and I have a couple of questions about it:

1. Isn't the `r` at the beginning of the string enough to make it a raw one?
2. If that's not the case, shouldn't you add an `r` to the last part of the string to also make it raw, just in case for the future?

**Comment by rear1019** [comment]
> I don't understand this change, so I have a couple of questions about it:
> 
>     1. Isn't the `r` at the beginning of the string enough to make it a raw one?
> 

No, it isn’t. String prefixes only apply to the immediate string literal. The syntax used here is called `String literal concatenation`, see [Python’s documenation](https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation) for details.

Technically, as far as the regular expression is concerned, the `r` is not required in the first line. However, it is required in the second.

File: spyder/plugins/pythonpath/utils.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def check_path(path):
     else:
         pattern = re.compile(
             r'.*(lib|lib64)/'
-            '(python|python\d+|python\d+\.\d+)/'
+            r'(python|python\d+|python\d+\.\d+)/'
             '(site|dist)-packages.*'
```
Comment (ccordoba12) [inline spyder/plugins/pythonpath/utils.py:26]:
```suggestion
            r'(site|dist)-packages.*'
```

**Review by ccordoba12 — COMMENTED** [review]
Just a small suggestion for you @juliangilbey.

File: spyder/plugins/pythonpath/utils.py (hunk 1 above)
Comment (juliangilbey) [reply spyder/plugins/pythonpath/utils.py]: Done!

**Review by ccordoba12 — APPROVED** [review]
Thanks @juliangilbey for your work and @rear1019 for reviewing it!

**Pull request merged**